app/services/folio_client.rb

```diff
@@ -118,20 +118,21 @@ def cancel_hold(resource, _item_key, patron_key)
   end
 
   # Cancel a hold request
-  # See https://s3.amazonaws.com/foliodocs/api/mod-patron/p/patron.html#patron_account__id__hold__holdid__cancel_post
   # @param [String] user_id the UUID of the user in FOLIO
   # @param [String] hold_id the UUID of the FOLIO hold
   def cancel_hold_request(user_id, hold_id)
-    # You would think FOLIO could look up this information and merge it itself, but no. You'd
-    # also think you could get the /circulation/requests/{id} endpoint data and use that,
-    # but the API schema is slightly different and OKAPI complains... so we have to look it
-    # up in the patron account data instead.
-    request_data = patron_account(user_id)['holds'].find { |h| h['requestId'] == hold_id }
+    # We formerly used the mod-patron API to cancel hold requests, but it is
+    # unable to cancel title level hold requests.
+    request_data = get_json("/circulation/requests/#{hold_id}")
+
+    # Ensure this is the user's request before trying to cancel it
+    request_data = {} unless request_data['requesterId'] == user_id
```

Comment:
I hate to even ask but... what about proxy requests?

Reply:
Looks like the proxy is tracked on the request via `proxyUserId`. I'm curious now to see whether this would have worked for the previous implementation with mod-patron because requests made on a behalf of a sponsor aren't returned on a proxy's account info.

Reply:
Ah, maybe it didn't. Should we file that away for later?